addons/mass_mailing/models/mailing.py

```diff
@@ -623,6 +623,17 @@ def _action_view_documents_filtered(self, view_filter):
             ),
         return action
 
+    def action_view_mailing_contacts(self):
+        """Show the mailing contacts who are in a mailing list selected for this mailing."""
+        self.ensure_one()
+        action = self.env['ir.actions.actions']._for_xml_id('mass_mailing.action_view_mass_mailing_contacts')
+        if self.contact_list_ids:
+            action['context'] = {
+                'default_subscription_list_ids': [(0, 0, {'list_id': self.contact_list_ids[0].id})],
+            }
+        action['domain'] = [('mailing_ids', 'in', self.ids)]
```

Comment:
Can't we set a domain on list_ids that is a real stored field instead ? Anyway we already access contact_list_ids just above ^^ .